src/PickerAvoidingView/index.ios.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { View } from 'react-native';
+import { PickerStateContext } from "../PickerStateProvider";
+import { IOS_MODAL_HEIGHT } from "../constants";
+
+const styles = {
+    container: {
+        flex: 1,
+        flexDirection: 'column',
+        justifyContent: 'flex-start',
+        alignItems: 'stretch',
+    },
+    top: {
+        flex: 1,
+    },
+    bottom: {
+        height: IOS_MODAL_HEIGHT,
+    },
+};
+
+/**
+ * PickerAvoidingView is a React component that adjusts the view layout to avoid
+ * being covered by an open iOS UIPickerView modal. It's meant to be similar to
+ * the built-in KeyboardAvoidingView component, but specifically tailored for
+ * iOS picker modals.
+ *
+ * In order for this component to work correctly, all the pickers and the
+ * PickerAvoidingView should have a PickerStateProvider ancestor.
+ *
+ * @param {React.ReactNode} props.children - The child components that should be
+ * protected from obstruction by the picker modal
+ */
+export function PickerAvoidingView(props) {
+    const context = React.useContext(PickerStateContext);
+    const isPickerOpen = context && context.isPickerOpen;
+
+    return (
+        <View style={styles.container}>
```

Comment:
replace this with prop value (props.style)

Reply:
But in order for this component to work, i..e to do what's it's meant to do, it has to have some specific styling, right? Picker modal is on the bottom, so int needs to be a column, etc... Do you mean that some styles should be forced, but other should be accepted from the user?

Reply:
my idea was to exactly replicate KeyboardAvoidingView.
Let's say it's like normal View which doesn't have any custom prop set.
So I meant this:
```suggestion
        <View style={props.style}>
```

Reply:
But this isn't "a normal view", this is a component that solves the picker obstruction problem by providing a column layout and adding a child that provides the extra space as needed.

This behavior shouldn't be overridable, from my perspective. It shouldn't be possible to make this.. a row, or something else.

I can see some value in reflecting `KeyboardAvoidingView`, but what's even more important is to make something that works and can't be broken easily via the public interface it exposes.

Reply:
I like this solution: https://github.com/facebook/react-native/blob/4fdad9e765c45b5c60ce25e671f430e03edc9ef6/packages/react-native/Libraries/Components/Keyboard/KeyboardAvoidingView.js#L228-L237
```
          <View
            ref={this.viewRef}
            style={StyleSheet.compose(style, {paddingBottom: bottomHeight})}
            onLayout={this._onLayout}
            {...props}>
            {children}
          </View>
```

Reply:
That's what I also figured out 😊

Reply:
ok let's try this, ignoring all previous comments and suggestions